### ISSUE TITLE
no error message when trying to execute a non-executable file

### ISSUE THREAD

**Issue by koo5** [fish-shell/fish-shell#5012]
fish, version 2.6.0
2.6.0
Linux localhost.localdomain 4.16.9-200.fc27.x86_64 #1 SMP Thu May 17 04:54:30 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
xterm-256color

>  sh -c 'env HOME=$(mktemp -d) fish'
```
koom@localhost ~/s/r/bin> sh -c 'env HOME=$(mktemp -d) fish'
                          
Welcome to fish, the friendly interactive shell
koom@localhost /h/k/s/r/bin> ~/sw/rdfconvert-0.4/bin/rdfconvert.sh
fish: Unknown command '/tmp/tmp.tqgucf9p7b/sw/rdfconvert-0.4/bin/rdfconvert.sh'
koom@localhost /h/k/s/r/bin>
```
fish fails without any error message when i try to run a script that doesnt have the execute flag:
```
koom@localhost ~/s/r/bin> chmod -x ~/sw/rdfconvert-0.4/bin/rdfconvert.sh
koom@localhost ~/s/r/bin> ~/sw/rdfconvert-0.4/bin/rdfconvert.sh
koom@localhost ~/s/r/bin> 
```

**Comment by faho** [comment]
Already fixed by 85e40540c904a8d16e8b6fcb46ec892ca98bf67 in 2.7.0. You'll want to upgrade.

**Comment by koo5** [comment]
thanks
